src/plugins/plot/src/inspector/PlotOptionsController.js

```diff
@@ -49,7 +49,10 @@ define([
     };
 
     PlotOptionsController.prototype.destroy = function () {
+        //untrack twice to offset strange timing issue
         configStore.untrack(this.configId);
```

Comment:
If I'm understanding this correctly what you've done in effect is make the PlotController ultimately responsible for the config store's lifecycle, which actually makes a lot of sense. I'm wondering whether we should just make that responsibility clearer by adding a `destroy` function to `ConfigStore` which would encapsulate the `destroy` behavior of `ConfigStore` and also allow it to be called manually from the `PlotOptionsController`. The tracking stuff could then just be removed from the `PlotController` and `PlotOptionsController`.

Reply:
This is a great idea. I changed it so that the plot controller is responsible for its config store and did away with the tracking system